src/types/pathways-model.d.ts

```diff
@@ -4,36 +4,36 @@ declare module 'pathways-model' {
     name: string;
     description?: string;
     library: string;
-    criteria: Criteria[];
-    states: {
-      [key: string]: GuidanceState | BranchState | State;
+    precondition: Precondition[];
+    nodes: {
+      [key: string]: PathwayActionNode | BranchNode | PathwayNode;
     };
     elm?: PathwayELM;
     // TODO: this should not be optional once we have the pathway builder
   }
 
   export interface PathwayELM {
     navigational?: object;
-    criteria?: object;
+    precondition?: object;
   }
 
   export interface EvaluatedPathway {
     pathway: Pathway;
     pathwayResults: PathwayResults | null;
   }
 
-  export interface Criteria {
+  export interface Precondition {
     elementName: string; // name of the mCODE element
     expected: string; // human readable value
     cql: string; // cql to fetch the value from a patient
   }
 
-  export interface State {
+  export interface PathwayNode {
     label: string;
     transitions: Transition[];
   }
 
-  export interface GuidanceState extends State {
+  export interface PathwayActionNode extends PathwayNode {
```

Comment:
Any reason this is `PathwayActionNode` and not just `ActionNode`?

Reply:
There was an error with importing both the ActionNode class and the ActionNode type in `Sidebar` in the pathways builder, so to fix it I had to rename the type.

Reply:
Let's rename that one ActionNodeEditor (or ActionNodeBuilder, or something along those lines). And similarly the BranchNode component in the builder should be renamed to BranchNodeEditor. That keeps those in sync, makes it clear that those components represent the editor of that type of node, rather than an instance of the node itself, and then this can go back to ActionNode, to align with BranchNode.